core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -4144,13 +4144,13 @@ protected RelNode convertInsert(SqlInsert call) {
         convertQueryRecursive(call.getSource(), true, targetRowType).project();
     RelNode massagedRel = convertColumnList(call, sourceRel);
 
-    return createModify(targetTable, massagedRel);
+    return createModify(targetTable, massagedRel, call.isOverwrite());
   }
 
   /**
    * Creates a relational expression to modify a table or modifiable view.
    */
-  private RelNode createModify(RelOptTable targetTable, RelNode source) {
+  private RelNode createModify(RelOptTable targetTable, RelNode source, boolean overwrite) {
```

Comment:
In general, I think we can avoid changing this class at all. I'd like to see us copy this class into our own code base into a similar location and make the changes there.

Normally, I'd say extend this class rather than copy it. I do believe that we're probably going to want to make some more substantial changes to this class in the near future so it might just be best if we copy it over now and begin to use it.

It appears like `SqlToRelConverter` can be overwritten to convert insert in whatever way we want. I'd say we want to subclass `SqlToRelConverter` and override the `convertInsert` method to be similar to this one. I think this will be our first instance of overwriting `SqlToRelConverter` so you might have to create that file.

If you're moving a file or creating a subclass, be sure to follow the same package path structure that the Calcite core uses. It will help us find and organize the classes.